vms/proposervm/vm_test.go

```diff
@@ -217,13 +217,22 @@ func initTestProposerVM(
 	return coreVM, valState, proVM, db
 }
 
-func waitForProposerWindow(vm *VM, chainTip snowman.Block, pChainHeight uint64) error {
+func (vm *VM) waitForProposerWindow() error {
+	ctx := context.Background()
+	preferred, err := vm.getBlock(ctx, vm.preferred)
+	if err != nil {
+		return fmt.Errorf("couldn't get preferred block: %w", err)
+	}
+
+	pChainHeight, err := preferred.pChainHeight(ctx)
+	if err != nil {
+		return fmt.Errorf("couldn't get P-Chain height from tip: %w", err)
+	}
+
 	var (
-		ctx              = context.Background()
-		childBlockHeight = chainTip.Height() + 1
-		parentTimestamp  = chainTip.Timestamp()
+		childBlockHeight = preferred.Height() + 1
+		parentTimestamp  = preferred.Timestamp()
```

Comment:
not related to this PR but since we're simplifying the function, shouldn't we just do:

```
		if now.After(delayUntil) {
			return nil
		}
```

instead of:

```
		if !now.Before(delayUntil) {
			return nil
		}
```

?